hr_course/models/hr_course.py

```diff
@@ -65,6 +76,62 @@ class HrCourse(models.Model):
         readonly=True,
     )
 
+    schedule_validity_end_date = fields.Date(
+        related="course_schedule_ids.validity_end_date"
+    )
+
+    course_attendee_ids = fields.One2many(
+        related="course_schedule_ids.course_attendee_ids"
+    )
+
+    @api.model
+    def send_course_notification_email(self):
```

Comment:
considering recently introduced refactoring should we notify not on "hr.course" but on "hr.course.schedule" model as it holds and attendees and validity_end_date?